tests/test_utils.py

```diff
@@ -517,26 +532,47 @@ def test_strip_registry_and_tag_from_image(img, expected):
     assert strip_registry_and_tag_from_image(img) == expected
 
 
-def test_checkout_git_repo(tmpdir):
-    repo_path = tmpdir.mkdir("repo").strpath
-    initialize_git_repo(repo_path)
-    with checkout_git_repo(repo_path, 'HEAD') as repo:
-        assert os.path.isdir(repo)
-        assert os.path.isdir(os.path.join(repo, '.git'))
-        assert subprocess.check_output(['git', 'status'], cwd=repo)
-
-    another_repo_path = tmpdir.mkdir("another_repo").strpath
-    initialize_git_repo(another_repo_path)
-    with pytest.raises(OsbsException) as e:
-        with checkout_git_repo(another_repo_path, 'bogusref') as another_repo:
-            assert os.path.isdir(another_repo)
-    assert 'Unable to reset branch' in e.value.message
-
-    non_repo_path = tmpdir.mkdir("notarepo").strpath
-    with pytest.raises(OsbsException) as e:
-        with checkout_git_repo(non_repo_path, 'HEAD', backoff_factor=0) as non_repo:
-            assert os.path.isdir(non_repo)
-    assert 'Unable to clone git repo' in e.value.message
+@pytest.mark.parametrize(('commit', 'branch', 'depth'), [
+    (None, None, None),
+    (TEST_DOCKERFILE_SHA1, None, None),
+    (TEST_DOCKERFILE_SHA1, TEST_DOCKERFILE_BRANCH, 1),
+    (TEST_DOCKERFILE_INIT_SHA1, TEST_DOCKERFILE_BRANCH, 1),
+    (TEST_DOCKERFILE_SHA1, None, 1),
+])
+@requires_internet
+def test_clone_git_repo(tmpdir, commit, branch, depth):
+    tmpdir_path = str(tmpdir.realpath())
+    repo_data = clone_git_repo(TEST_DOCKERFILE_GIT, tmpdir_path, commit=commit,
+                               branch=branch, depth=depth)
+    assert repo_data.commit_id is not None
+    assert tmpdir_path == repo_data.repo_path
+    if commit:
+        assert commit == repo_data.commit_id
+    assert len(repo_data.commit_id) == 40  # current git hashes are this long
+    assert os.path.isdir(os.path.join(tmpdir_path, '.git'))
+
+
+@pytest.mark.parametrize(('commit', 'branch', 'depth'), [
+    ("bad", None, None),
+    ("bad", TEST_DOCKERFILE_BRANCH, 1),
+    ("bad", TEST_DOCKERFILE_BRANCH, 1),
+    ("bad", None, 1),
+])
+@requires_internet
+def test_clone_git_repo_commit_failure(tmpdir, commit, branch, depth):
+    tmpdir_path = str(tmpdir.realpath())
+    with pytest.raises(OsbsException) as exc:
+        clone_git_repo(TEST_DOCKERFILE_GIT, tmpdir_path, commit=commit, retry_times=1,
+                       branch=branch, depth=depth)
+    assert 'cannot find commit' in exc.value.message
+
+
+@requires_internet
+def test_clone_git_repo_total_failure(tmpdir):
```

Comment:
This test is prompting a GitHub username. Travis times out here.

Reply:
Fixed.